src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -44,188 +54,1324 @@ const ActorInit En_Dekubaba_InitVars = {
+        if (this->timer == 2) {
+            this->timer = 40;
+        } else {
+            this->timer = 40;
+        }
+    }
+
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.y = this->actor.home.pos.y + (60.0f * this->size);
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    this->actionFunc = EnDekubaba_StunnedVertical;
+}
+
+void EnDekubaba_StunnedVertical(EnDekubaba* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 0) {
+        EnDekubaba_DisableHitboxes(this);
+
+        if (this->actor.xzDistToPlayer < (80.0f * this->size)) {
+            EnDekubaba_SetupPrepareLunge(this);
+        } else {
+            EnDekubaba_SetupRecover(this);
+        }
+    }
+}
+
+void EnDekubaba_SetupSway(EnDekubaba* this) {
+    this->targetSwayAngle = -0x6000;
+    this->stemSectionAngle[2] = -0x5000;
+    this->stemSectionAngle[1] = -0x4800;
+    EnDekubaba_DisableHitboxes(this);
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 35);
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actionFunc = EnDekubaba_Sway;
+}
+
+/**
+ * Sway back and forth with decaying amplitude until close enough to vertical.
+ */
+void EnDekubaba_Sway(EnDekubaba* this, PlayState* play) {
+    s16 angleToVertical;
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_ScaledStepToS(&this->actor.shape.rot.x, this->stemSectionAngle[0], 0x71C);
+    Math_ScaledStepToS(this->stemSectionAngle, this->stemSectionAngle[1], 0x71C);
+    Math_ScaledStepToS(&this->stemSectionAngle[1], this->stemSectionAngle[2], 0x71C);
+    if (Math_ScaledStepToS(&this->stemSectionAngle[2], this->targetSwayAngle, 0x71C) != 0) {
+        this->targetSwayAngle = -0x4000 - (s32)((this->targetSwayAngle + 0x4000) * 0.8f);
+    }
+
+    angleToVertical = this->targetSwayAngle + 0x4000;
+
+    if (ABS_ALT(angleToVertical) < 0x100) {
+        this->collider.base.acFlags |= AC_ON;
+        if (this->actor.xzDistToPlayer < (80.0f * this->size)) {
+            EnDekubaba_SetupPrepareLunge(this);
+        } else {
+            EnDekubaba_SetupRecover(this);
+        }
+    }
+    EnDekubaba_UpdateHeadPosition(this);
+}
+
+void EnDekubaba_SetupFrozen(EnDekubaba* this) {
+    EnDekubaba_SetFrozenEffects(this);
+    this->actionFunc = EnDekubaba_Frozen;
+}
+
+void EnDekubaba_Frozen(EnDekubaba* this, PlayState* play) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        EnDekubaba_SpawnIceEffects(this, play);
+
+        if (this->actor.colChkInfo.health == 0) {
+            EnDekubaba_SetupShrinkDie(this);
+        } else if (this->actor.xzDistToPlayer < (80.0f * this->size)) {
+            EnDekubaba_SetupPrepareLunge(this);
+        } else {
+            EnDekubaba_SetupRecover(this);
+        }
+    }
+}
+
+void EnDekubaba_SetupDeadStickDrop(EnDekubaba* this, PlayState* play) {
```

Comment:
There's a big difference here: when it drops a Nut the Deku Baba just shrinks and despawns. When it drops a stick, it turns into the long stick that stands upright on the ground, which is what that actionfunction will do.

Reply:
I would still say that Prune and Wither fit there.

Reply:
I'm distinguishing between the part where it draws the original Baba model in two segments (this being pruned), and drawing the completely different stick model, I think it's sensible to describe the state with the different model completely differently.

Reply:
I think naming after the drop is better imo

Reply:
Personally I think it is fine to name this part based on how its model is changed, so `DeadStickDrop` is fine. However it is weird that there is no analogous `PruneDie` to `ShrinkDie`. IMO, `PrunedSomersault` should be renamed to `PrunedDie`.

Reply:
I think I'll do `PrunedSomersaultDie` since the more notable thing about the action is the way it vaults over after the disconnection. I can see the point of including `Die` although it sits a little uncomfortably alongside the way it just disappears immediately in the other case.

Reply:
Why the preference for naming based on what is dropped over what the deku baba (this actor) is doing?

Beyond that: It's still inconsistent. One is `ShrinkDie` and the other is `EnDekubaba_DeadStickDrop`?